tests/test_fastavro.py

```diff
@@ -2649,3 +2649,14 @@ def test_schema_migration_should_fail_with_different_names():
 
     with pytest.raises(SchemaResolutionError):
         roundtrip(writer_schema, records, reader_schema=reader_schema)
+
+
+def test_union_of_float_and_double_keeps_precision():
+    """https://github.com/fastavro/fastavro/issues/437"""
+    schema = ["float", "double"]
```

Comment:
Could you change this line to the following so that it covers the case where we have found a `float` and we are continuing to look through the schemas but find something other than double?

```suggestion
    schema = ["float", "string", "double"]
```